handlers/changes.js

```diff
@@ -178,19 +192,48 @@ var getChanges = function(feed) {
     cleanUp(feed);
     if (err) {
       feed.res.write(error(503, 'Error processing your changes'));
-    } else if (!changes || !changes.results) {
+    } else {
+      const changes = mergeChangesResults(results);
+      if (changes) {
+        prepareResponse(feed, changes);
+      } else {
+        feed.res.write(error(503, 'No _changes error, but malformed response.'));
+      }
+    }
+    feed.res.end();
+  });
+};
+
+const mergeChangesResults = results => {
+  const merged = {
+    results: [],
+    last_seq: 0, // TODO in couch 2, this will not be numeric
+  };
+  let err = false;
+
+  results.forEach(changes => {
+    if (err) {
+      return;
+    }
+    if (!changes || !changes.results) {
       // See: https://github.com/medic/medic-webapp/issues/3099
       // This should never happen, but apparently it does sometimes.
       // Attempting to log out the response usefully to see what's occuring
-      var malformedChangesError = 'No _changes error, but malformed response: ';
-      var printableChanges = JSON.stringify(changes);
-      console.error(malformedChangesError, printableChanges);
-      feed.res.write(error(503, malformedChangesError + printableChanges));
+      console.error('No _changes error, but malformed response:', JSON.stringify(changes));
+      err = true;
     } else {
-      prepareResponse(feed, changes);
+      changes.results.forEach(r => merged.results.push(r));
+      merged.last_seq = Math.max(merged.last_seq, changes.last_seq);
     }
-    feed.res.end();
   });
+
+  if (err) {
+    return;
+  }
+
+  merged.results.sort((a, b) => a.seq - b.seq);
```

Comment:
Probably want another todo here about couchdb 2.0. It will mostly be the same, just need to cast to an int and get rid of the string after the number